### PR TITLE
Login admin

### PR DESCRIPTION
**Pull request by yaroslavbilokin**

### CODE REVIEW

File: modules/news/news.service.js [hunk 1]
```diff
@@ -14,9 +14,6 @@ class NewsService {
   }
 
   async updateNews(id, news) {
-    if (await this.checkNewsExist(news)) {
```
Comment (Alan18081) [inline modules/news/news.service.js:17]: Why have you removed this method call? It is not related to the your merge request, please, revert it

File: modules/user/user.service.js [hunk 2]
```diff
@@ -105,6 +105,45 @@ class UserService {
...
+
+    const user = await this.getUserByFieldOrThrow('email', email);
+
+    const match = await bcrypt.compare(
+      password,
+      user.credentials.find(cred => cred.source === 'horondi').tokenPass,
+    );
+
+    if (user.role === 'user') {
+      throw new AuthenticationError(`No access!`);
```
Comment (Alan18081) [inline modules/user/user.service.js:128]: Please, use the uppercased error message as it is done for the news, for example

File: modules/user/user.service.js [hunk 3]
```diff
@@ -105,6 +105,45 @@ class UserService {
...
+    }
+
+    if (!match) {
+      throw new AuthenticationError(`Wrong password!`);
```
Comment (Alan18081) [inline modules/user/user.service.js:132]: Please, use the uppercased error message as it is done for the news, for example

File: modules/user/user.service.js [hunk 4]
```diff
@@ -105,6 +105,45 @@ class UserService {
...
+    }
+
+    const token = generateToken(user._id, user.email);
+
+    return {
+      user: {
+        ...user._doc,
+      },
+      _id: user._id,
+      role: user.role,
```
Comment (Alan18081) [inline modules/user/user.service.js:142]: I think `role` is returned as part of the `user`. Why do we need to duplicate it here?

File: typeDefs.js [hunk 5]
```diff
@@ -273,6 +273,7 @@ const typeDefs = gql`
     "User Mutation"
     registerUser(user: userRegisterInput!): User
     loginUser(user: userLoginInput!): User
+    loginAdmin(user: UserInput!): User
```
Comment (Alan18081) [inline typeDefs.js:276]: Please, rename the argument `user` to `loginInput` because it is data related to the login not the user itself. Also, please do it for the `loginUser` too

File: modules/user/user.service.js [hunk 6]
```diff
@@ -105,6 +105,45 @@ class UserService {
...
+      throw new UserInputError('Errors', {
+        errors,
+      });
+    }
+
+    const user = await this.getUserByFieldOrThrow('email', email);
+
+    const match = await bcrypt.compare(
+      password,
+      user.credentials.find(cred => cred.source === 'horondi').tokenPass,
```
Comment (Alan18081) [inline modules/user/user.service.js:124]: Please, move the string `horondi` to some enum, create object named `SOURCES` and store value `horondi` there to ensure that no typo can happen

File: modules/user/user.service.js [hunk 7]
```diff
@@ -105,6 +105,45 @@ class UserService {
...
+    );
+
+    if (user.role === 'user') {
```
Comment (Alan18081) [inline modules/user/user.service.js:127]: Please, move the string `user` to some enum, create object named ROLES and store value `user` there to ensure that no typo can happen

File: modules/user/user.service.js (hunk 2 above)
Comment (yaroslavbilokin) [reply modules/user/user.service.js:128]: Changed!

File: modules/user/user.service.js (hunk 3 above)
Comment (yaroslavbilokin) [reply modules/user/user.service.js:132]: Changed!

File: modules/user/user.service.js (hunk 4 above)
Comment (yaroslavbilokin) [reply modules/user/user.service.js:142]: You're right! Removed!

File: modules/user/user.service.js (hunk 7 above)
Comment (yaroslavbilokin) [reply modules/user/user.service.js:127]: Moved!

File: modules/user/user.service.js (hunk 6 above)
Comment (yaroslavbilokin) [reply modules/user/user.service.js:124]: Moved!

File: typeDefs.js (hunk 5 above)
Comment (yaroslavbilokin) [reply typeDefs.js:276]: Renamed!

File: modules/news/news.service.js (hunk 1 above)
Comment (yaroslavbilokin) [reply modules/news/news.service.js:17]: Changed!

File: error-messages/user.messages.js [hunk 8]
```diff
@@ -4,6 +4,7 @@ const INPUT_NOT_VALID = 'INPUT_NOT_VALID';
 const WRONG_CREDENTIALS = 'WRONG_CREDENTIALS';
 const USER_NOT_AUTHORIZED = 'USER_NOT_AUTHORIZED';
 const EMAIL_ERROR = 'EMAIL_ERROR';
+const NO_ACCESS = 'NO_ACCESS';
```
Comment (Alan18081) [inline error-messages/user.messages.js:7]: Please, rename it to the `INVALID` permissions, it is much easier to understand, thanks

File: modules/news/news.service.js [hunk 9]
```diff
@@ -36,7 +41,7 @@ class NewsService {
         },
       },
     });
-    return newsCount > 0;
+    return newsCount;
```
Comment (Alan18081) [inline modules/news/news.service.js:44]: Revert this change, because this method performs check and returns a boolean value but not the count

File: typeDefs.js [hunk 10]
```diff
@@ -278,9 +278,9 @@ const typeDefs = gql`
     updateNews(id: ID!, news: NewsInput!): NewsResult
 
     "User Mutation"
-    registerUser(user: userRegisterInput!): User
-    loginUser(user: userLoginInput!): User
-    loginAdmin(user: UserInput!): User
+    registerUser(user: UserInput!, language: Int!): User
+    loginUser(loginInput: userLoginInput!): User
```
Comment (Alan18081) [inline typeDefs.js:282]: Please, rename `userLoginInput` to `LoginInput`, there is no need to include word `user`

File: typeDefs.js [hunk 11]
```diff
@@ -278,9 +278,9 @@ const typeDefs = gql`
...
+    loginAdmin(loginInput: userLoginInput!): User
```
Comment (Alan18081) [inline typeDefs.js:283]: Please, rename `userLoginInput` to `LoginInput`, there is no need to include word `user`

Comment (yaroslavbilokin) [reply typeDefs.js:283]: Renamed!

File: typeDefs.js (hunk 10 above)
Comment (yaroslavbilokin) [reply typeDefs.js:282]: Renamed!

File: error-messages/user.messages.js (hunk 8 above)
Comment (yaroslavbilokin) [reply error-messages/user.messages.js:7]: Renamed!

File: modules/news/news.service.js (hunk 9 above)
Comment (yaroslavbilokin) [reply modules/news/news.service.js]: Changed!
